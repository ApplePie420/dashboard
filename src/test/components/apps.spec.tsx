import { render } from "@testing-library/react";
import {
  App,
  AppCategory,
  AppList,
  IAppProps,
  IAppCategoryProps,
  IAppListProps,
} from "../../components/apps";

const appProps: IAppProps = {
  name: "App Test",
  icon: "bug_report",
  url: "#",
  displayURL: "test",
};

const appCategoryProps: Array<IAppCategoryProps> = [
  {
    name: "Test",
    items: [appProps, appProps],
  },
  {
    name: "",
    items: [appProps, appProps],
  },
];

const appListProps: Array<IAppListProps> = [
  {
    categories: appCategoryProps,
    apps: [appProps, appProps],
  },
  {
    apps: undefined,
    categories: appCategoryProps,
  },
  {
    apps: [appProps, appProps],
    categories: undefined,
  },
  {
    apps: undefined,
    categories: undefined,
  },
];

describe("app.tsx", () => {
  it("Tests App rendering with newTab", () => {
    const tests = [true, false];

    tests.forEach((test: boolean) => {
      const { asFragment } = render(
        <App
          name={appProps.name}
          icon={appProps.icon}
          url={appProps.url}
          displayURL={appProps.displayURL}
          newTab={test}
        />,
      );

      expect(asFragment).toMatchSnapshot();
    });
  });

  it("Tests App rendering without newTab", () => {
    const { asFragment } = render(
      <App
        name={appProps.name}
        icon={appProps.icon}
        url={appProps.url}
        displayURL={appProps.displayURL}
      />,
    );

    expect(asFragment).toMatchSnapshot();
  });

  it("Tests AppCategory rendering", () => {
    appCategoryProps.forEach((appCategory) => {
      const { asFragment } = render(
        <AppCategory name={appCategory.name} items={appCategory.items} />,
      );

      expect(asFragment).toMatchSnapshot();
    });
  });

  it("Tests AppList rendering", () => {
    appListProps.forEach((appList) => {
      const { asFragment } = render(
        <AppList apps={appList.apps} categories={appList.categories} />,
      );

      expect(asFragment).toMatchSnapshot();
    });
  });

  it("Tests AppList rendering without any props", () => {
    const { asFragment } = render(<AppList />);
    expect(asFragment).toMatchSnapshot();
  });
});
