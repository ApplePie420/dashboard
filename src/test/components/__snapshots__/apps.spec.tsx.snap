// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app.tsx Tests App rendering with newTab 1`] = `[Function]`;

exports[`app.tsx Tests App rendering with newTab 2`] = `[Function]`;

exports[`app.tsx Tests App rendering without newTab 1`] = `[Function]`;

exports[`app.tsx Tests AppCategory rendering 1`] = `[Function]`;

exports[`app.tsx Tests AppCategory rendering 2`] = `[Function]`;

exports[`app.tsx Tests AppList rendering 1`] = `[Function]`;

exports[`app.tsx Tests AppList rendering 2`] = `[Function]`;

exports[`app.tsx Tests AppList rendering 3`] = `[Function]`;

exports[`app.tsx Tests AppList rendering 4`] = `[Function]`;

exports[`app.tsx Tests AppList rendering without any props 1`] = `[Function]`;
